handlers/whatsapp/whatsapp.go

```diff
@@ -189,55 +333,150 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
 	sendPath, _ := url.Parse("/v1/messages")
 	sendURL := url.ResolveReference(sendPath).String()
 
-	// TODO: figure out sending media
+	mediaPath, _ := url.Parse("/v1/media")
+	mediaURL := url.ResolveReference(mediaPath).String()
 
 	status := h.Backend().NewMsgStatusForID(msg.Channel(), msg.ID(), courier.MsgErrored)
-	parts := handlers.SplitMsg(msg.Text(), maxMsgLength)
-	for i, part := range parts {
-		payload := mtPayload{
-			To:   msg.URN().Path(),
-			Type: "text",
-		}
-		payload.Text.Body = part
 
-		jsonBody, err := json.Marshal(payload)
+	// TODO: figure out sending media
+	if len(msg.Attachments()) > 1 {
+		duration := time.Now().Sub(start)
+		err = fmt.Errorf("message has %d attachments", len(msg.Attachments()))
+		courier.NewChannelLogFromError("WhatsApp only allows for a single attachment to a message.", msg.Channel(), msg.ID(), duration, err)
+		return status, err
+
+	} else if len(msg.Attachments()) == 1 {
+
+		attachment := msg.Attachments()[0]
+		parts := strings.SplitN(attachment, ":", 2)
+		mimeType := parts[0]
+		s3url := parts[1]
+
+		// retrieve the media to be sent from S3
+		req, _ := http.NewRequest(http.MethodGet, s3url, nil)
+		s3rr, err := utils.MakeHTTPRequest(req)
```

Comment:
I had just mentioned that on Slack, here we know we can allow all content types so we do something like 
https://github.com/nyaruka/courier/blob/master/backends/rapidpro/msg.go#L215

And keep the `isText` flag in `newRRFromResponse`

Reply:
I would also make a function that handle one attachment work here, download from s3 and upload to WA that will return the mediaId that can be used in the the payload below